src/arrayMethodReduce.js

```diff
@@ -1,12 +1,18 @@
-'use strict';
-
-/**
- * Implement method Reduce
- */
-function applyCustomReduce() {
-  [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
-  };
-}
-
-module.exports = applyCustomReduce;
+'use strict';
+
+/**
+ * Implement method Reduce
+ */
+function applyCustomReduce() {
+  [].__proto__.reduce2 = function(callback, initialValue) {
+    let acc = arguments.length === 1 ? this[0] : initialValue;
+
+    for (let i = arguments.length === 1 ? 1 : 0; i < this.length; i++) {

```

Comment:
It's better to move `startIndex` to the separate constant. It should looks better